experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -184,6 +186,49 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 		this.context.insertNodes(fieldPath, index, newContent);
 	}
 
+	public moveNodes(
+		sourceIndex: number,
+		count: number,
+		destinationIndex: number,
+		destinationField?: EditableField,
+	): void {
+		const sourceFieldPath = this.cursor.getFieldPath();
+		const destinationFieldKindIdentifier =
+			destinationField !== undefined
+				? destinationField.fieldSchema.kind.identifier
+				: this.fieldSchema.kind.identifier;
+
+		assert(
+			this.fieldSchema.kind.identifier === sequence.identifier &&
+				destinationFieldKindIdentifier === sequence.identifier,
+			"Both source and destination fields must be sequence fields.",
+		);
+
+		const destinationFieldProxy =
+			destinationField !== undefined
+				? (destinationField[proxyTargetSymbol] as ProxyTarget<Anchor | FieldAnchor>)
+				: this;
+
+		assert(
+			destinationFieldProxy === undefined ||
```

Comment:
Now that this is using destinationFieldProxy, we should always have a destination (even in the case where its this), so I don't think we need the `destinationFieldProxy === undefined ||` anymore, right?

Reply:
replaced with assertValidIndex function